src/backend/optimizer/path/allpaths.c

```diff
@@ -1699,10 +1699,14 @@ set_subquery_pathlist(PlannerInfo *root, RelOptInfo *rel,
 		 * from a bunch of conditions that can cause motion in the plan,
 		 * we can only gather it to singleQE and materialize the data because we
 		 * cannot pass params across motion.
+		 * We also set can_have_dependencies to disable other related optimizations.
 		 */
 		if ((!bms_is_empty(required_outer)) &&
 			check_query_for_possible_motion(subquery))
+		{
 			config->force_singleQE = true;
+			config->can_have_dependencies = true;
```

Comment:
Why not simple
```suggestion
			root->is_correlated_subplan = true;
```

Reply:
The main reason for this patch is to disable MIN/MAX optimizations. This won't disable the optimization for that specific subplan (only for child subplans). Also, it is not guaranteed that a lateral join has correlations, it is possible to just add a `lateral` keyword to any join without dependency on previous tables.